ddpui/ddpdbt/elementary_service.py

```diff
@@ -0,0 +1,397 @@
+    elementary_profile["elementary"]["outputs"][target] = dbt_profile[target]["outputs"][target]
+
+    # set schema to elementary
+    elementary_profile["elementary"]["outputs"][target]["schema"] = "elementary"
+
+    elementary_profile_dir = Path(dbt_project_params.project_dir) / "elementary_profiles"
+
+    if not elementary_profile_dir.exists():
+        elementary_profile_dir.mkdir()
+
+    elementary_profile_file = elementary_profile_dir / "profiles.yml"
+    with open(elementary_profile_file, "w", encoding="utf-8") as elementary_profile_file_f:
+        yaml.dump(elementary_profile, elementary_profile_file_f)
+
+    logger.info("wrote elementary profile to %s", elementary_profile_file)
+
+    return {"status": "success"}
+
+
+def fetch_elementary_report(org: Org):
+    """fetch previously generated elementary report"""
+    if org.dbt is None:
+        return "dbt is not configured for this client", None
+
+    project_dir = Path(DbtProjectManager.get_dbt_project_dir(org.dbt))
+
+    if not os.path.exists(project_dir / "elementary_profiles"):
+        return "set up elementary profile first", None
+
+    s3 = boto3.client(
+        "s3",
+        "ap-south-1",
+        aws_access_key_id=os.getenv("ELEMENTARY_AWS_ACCESS_KEY_ID"),
+        aws_secret_access_key=os.getenv("ELEMENTARY_AWS_SECRET_ACCESS_KEY"),
+    )
+    bucket_file_path = make_edr_report_s3_path(org)
+    try:
+        s3response = s3.get_object(
+            Bucket=os.getenv("ELEMENTARY_S3_BUCKET"),
+            Key=bucket_file_path,
+        )
+        logger.info("fetched s3response")
+    except boto3.exceptions.botocore.exceptions.ClientError:
+        return "report has not been generated", None
+    except Exception:
+        return "error fetching elementary report", None
+
+    report_html = s3response["Body"].read().decode("utf-8")
+    htmlfilename = str(project_dir / "elementary-report.html")
+    with open(htmlfilename, "w", encoding="utf-8") as indexfile:
+        indexfile.write(report_html)
+        indexfile.close()
+    logger.info("wrote elementary report to %s", htmlfilename)
+
+    redis = RedisClient.get_instance()
+    token = uuid4()
+    redis_key = f"elementary-report-{token.hex}"
+    redis.set(redis_key, htmlfilename.encode("utf-8"), 600)
+    logger.info("created redis key %s", redis_key)
+
+    return None, {
+        "token": token.hex,
+        "created_on_utc": s3response["LastModified"].isoformat(),  # e.g. 2024-06-07T00:44:08+00:00
+        "created_on_ist": as_ist(
+            s3response["LastModified"]
+        ).isoformat(),  # e.g. 2024-06-07T06:14:08+05:30
+    }
+
+
+def refresh_elementary_report_via_prefect(orguser: OrgUser) -> dict:
+    """refreshes the elementary report for the current date using the prefect deployment"""
+    org: Org = orguser.org
+    odf = OrgDataFlowv1.objects.filter(
+        org=org, name__startswith=f"pipeline-{org.slug}-generate-edr"
+    ).first()
```

Comment:
we have a task of type `edr`. We should use this & the mapping in DataFlowOrgTask to this task to get the dataflow.

the name is an arbitrary one and can change anytime.